### PR TITLE
RavenDB-4823 WIP

### PR DESCRIPTION
**Pull request by myarichuk**
* add a class that can query local endpoint without going through http
* add endpoints for both per database and server-wide debug-info output
* also refactor serveral debug endpoints so they would return blittable object and not blittable array - proper parsing of results
(RavenDB-4823)

### CODE REVIEW

File: src/Raven.Server/Documents/Handlers/Debugging/TransactionDebugHandler.cs [hunk 1]
```diff
@@ -39,7 +38,10 @@ public Task TxInfo()
             using (ContextPool.AllocateOperationContext(out context))
             using (var writer = new BlittableJsonTextWriter(context, ResponseBodyStream()))
             {
-                context.Write(writer, ToJson(results));
+                context.Write(writer, new DynamicJsonValue
+                {
+                    ["tx-info"] = ToJson(results)
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Handlers/Debugging/TransactionDebugHandler.cs:43]: maybe TxInfo would be better?

File: src/Raven.Server/Documents/Handlers/ReplicationHandler.cs [hunk 2]
```diff
@@ -398,7 +406,10 @@ public Task GetReplicationReconnectionQueue()
                     });
                 }
 
-                context.Write(writer, data);
+                context.Write(writer, new DynamicJsonValue
+                {
+                    ["Queue-Info"] = data
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Handlers/ReplicationHandler.cs:411]: Maybe remove '-' ? Those tend to be problematic.

File: src/Raven.Server/Documents/Handlers/Debugging/DatabaseDebugInfoPackageHandler.cs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.Handlers.Debugging
+{
+    public class DatabaseDebugInfoPackageHandler : DatabaseRequestHandler
+    {
+        [RavenAction("/databases/*/debug/info-package", "GET", IsDebugInformationEndpoint = true)]
+        public async Task GetInfoPackage()
+        {
+            var contentDisposition = $"attachment; filename=debug-info of {Database.Name} {DateTime.UtcNow}.zip";
```
Comment (ayende) [inline src/Raven.Server/Documents/Handlers/Debugging/DatabaseDebugInfoPackageHandler.cs:19]: `Database debug-info ..`

File: src/Raven.Server/Documents/Handlers/Debugging/TransactionDebugHandler.cs (hunk 1 above)
Comment (ayende) [reply src/Raven.Server/Documents/Handlers/Debugging/TransactionDebugHandler.cs:43]: `TransactionInformation` would be better

File: src/Raven.Server/ServerWide/LocalEndpointClient.cs [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+using Raven.Server.Routing;
+using Raven.Server.Web;
+using Sparrow.Json;
+
+namespace Raven.Server.ServerWide
+{
+    /// <summary>
+    /// This class is useful for interacting with local endpoints without going through HTTP
+    /// </summary>
+    public class LocalEndpointClient
```
Comment (ayende) [inline src/Raven.Server/ServerWide/LocalEndpointClient.cs:26]: This is nice

**Pull request merged**